src/shared/dlt_config_file_parser.c

```diff
@@ -206,28 +196,35 @@ static int dlt_config_file_set_section_data(DltConfigFile *file, char *str1, cha
     /* copy data into structure */
     strncpy(&s->keys[key_number * DLT_CONFIG_FILE_ENTRY_MAX_LEN], str1, DLT_CONFIG_FILE_ENTRY_MAX_LEN);
 
-    item.key = strdup(str1);
-    item.data = strdup(str2);
-
-    /* check if already stored */
-    if (hsearch_r(item, FIND, &ret, &s->data) == 0)
+    if (s->list == NULL)
```

Comment:
missing a brief description what you are doing..
Looks like list doens't exists at all here?